src/common/transformations/src/transformations/op_conversions/eye_decomposition.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/op_conversions/eye_decomposition.hpp"
+
+#include <memory>
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/util/gather_nd_base.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/opsets/opset8.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+namespace ov {
+namespace pass {
+
+/** \brief Check if output is rank one and data type can be i32 or i64. */
+const auto is_rank_one_int_shape = [](const Output<Node>& output) -> bool {
+    return pattern::type_matches_any({element::i32, element::i64})(output) && pattern::rank_equals(1)(output);
+};
+
+/** \brief Predicate to check eye k node is valid. */
+const auto k_predicate = [](const Output<Node>& output) -> bool {
+    return is_rank_one_int_shape(output) && (output.get_partial_shape()[0].get_length() == 1);
+};
+
+/** \brief Predicate to check eye batch node is valid. */
+const auto batch_predicate = [](const Output<Node>& output) -> bool {
+    return is_rank_one_int_shape(output) && output.get_partial_shape()[0].get_length();
+};
+
+std::shared_ptr<Node> EyeDecomposition::make_eye_model(const Output<Node>& height,
+                                                       const Output<Node>& width,
+                                                       const Output<Node>& k,
+                                                       element::Type dtype) {
+    const auto zero_int = register_new_node(op::v0::Constant::create(element::i64, Shape{1}, {0}));
+    const auto zero = register_new_node(op::v0::Constant::create(dtype, Shape{1}, {0}));
+    const auto one = register_new_node(op::v0::Constant::create(dtype, ov::Shape{1}, {1}));
+
+    const auto k_neg = register_new_node<op::v0::Negative>(k);
+    const auto k_axis = register_new_node<op::v0::Concat>(OutputVector{k_neg, k}, 0);
+
+    const auto eye_shape = register_new_node<op::v0::Concat>(OutputVector{height, width}, 0);
+
+    // Calculate eye zero padding and internal square eye size.
+    const auto pad_start =
+        register_new_node<op::v1::Minimum>(eye_shape, register_new_node<op::v1::Maximum>(zero_int, k_axis));
+    const auto shape_pad_diff = register_new_node<op::v1::Subtract>(eye_shape, pad_start);
+    const auto eye_size = register_new_node<op::v1::ReduceMin>(shape_pad_diff, zero_int, true);
```

Comment:
The good practise is using ops from the same opset, preferably with the newest one (now it is `opset9`

Reply:
Used opset9.